x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/inference/persistence/TrainedModelProvider.java

```diff
@@ -272,11 +272,10 @@ public void getTrainedModel(final String modelId, final boolean includeDefinitio
             listener::onFailure
         );
 
-        executeAsyncWithOrigin(client,
-            ML_ORIGIN,
+        MultiSearchResponse response = client.execute(
             MultiSearchAction.INSTANCE,
-            multiSearchRequestBuilder.request(),
-            multiSearchResponseActionListener);
+            multiSearchRequestBuilder.request()).actionGet();
```

Comment:
This is the same problem we have with the FetchSubPhase PR, we have make a blocking call somewhere